src/fenics_constitutive/models/elasticity_laws.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+from typing import Protocol
+
+import numpy as np
+
+from fenics_constitutive import StressStrainConstraint
+
+from .utils import lame_parameters
+
+
+class ElasticityLaw(Protocol):
+    def get_D(self, E: float, nu: float) -> np.ndarray: ...
+    def get_I2(self, stress_strain_dim: int) -> np.ndarray: ...
+
+
+class FullConstraintLaw(ElasticityLaw):
```

Comment:
These laws are all elastic, for another model such as Mises plasticity another protocol is developed. As such, I would suggest to reflect that also in the name, so e.g. ElasticityFullConstraintLaw. (We could also add Lin, but currently I do not see an nonlinear elastic law to be implemented in the future), same with the subsequent names.